agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import numpy.typing as npt
+from typing_extensions import override
+
+import pinocchio as pin
+
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryPoint
+
+
+class WarmStartReference(WarmStartBase):
+    """
+    A class for generating warmstart values for trajectory optimization problem.
+
+    This class uses a reference trajectory and the robot model to compute the initial state,
+    state vectors, and control inputs.
+
+    Attributes:
+    _rmodel (pin.Model | None): The robot's Pinocchio model, used for forward dynamics computations.
+    _rdata (pin.Data | None): Data structure associated with the Pinocchio model.
+
+    Methods:
+        setup(rmodel: pin.Model) -> None:
+            Initializes the robot model and its associated data structure for later computations.
+
+        generate(
+            initial_state: TrajectoryPoint,
+            reference_trajectory: list[TrajectoryPoint],
+        ) -> tuple[
+            npt.NDArray[np.float64],
+            list[npt.NDArray[np.float64]],
+            list[npt.NDArray[np.float64]],
+        ]:
+            Generates the initial state, reference state vectors, and control inputs for warmstart.
+
+            Parameters:
+                initial_state (TrajectoryPoint): The starting state of the robot, containing joint configuration
+                    and velocity information.
+                reference_trajectory (list[TrajectoryPoint]): A sequence of desired trajectory points, each containing
+                    joint configuration, velocity, and acceleration.
+
+            Returns:
+                tuple: A tuple containing:
+                    - x0 (npt.NDArray[np.float64]): The initial state vector `[q, v]` where `q` is the joint configuration
+                    and `v` is the joint velocity.
+                    - x_init (list[npt.NDArray[np.float64]]): A list of state vectors `[q, v]` constructed from the
+                    reference trajectory.
+                    - u_init (list[npt.NDArray[np.float64]]): A list of control inputs computed using inverse dynamics
+                    (RNEA) based on the reference trajectory.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._rmodel: pin.Model | None = None
+        self._rdata: pin.Data | None = None
+
+    def setup(self, rmodel: pin.Model) -> None:
+        self._rmodel = rmodel
+        self._rdata = self._rmodel.createData()
```

Comment:
Do we assume that warmstart in general doesn't need pinocchio? This function generally breaks compatibility with the abstract class so is there a good explanation why the base class dosn't have `setup` function?

Reply:
hmmm good questions. I think it makes sense to assume it doesn't need Pinocchio, maybe we warm start with zeros and then we just need dimensions. Or we warmstart from the previous solution. But sure, we can add setup to the base class

Reply:
I added it to the base class